### ISSUE TITLE
API 4.7

### ISSUE THREAD

**Issue by Bibo-Joshi** [python-telegram-bot/python-telegram-bot#1856]
Pure API updates:

- [x] Add class `Dice`
- [x] Add `Bot.send_dice`
- [x] Add `Message.dice`
- [x] Add `Message.reply_dice`
- [x] Add class `BotCommand`
- [x] Add `Bot.get_my_commands`
- [x] Add `Bot.set_my_commands`
- [x] Add `Bot.commands` property
- [x] Add parameter `tgs_sticker` to `Bot.create_new_sticker_set`
- [x] Add parameter `tgs_sticker` to `Bot.add_sticker_to_set`
- [x] Add `StickerSet.thumb`
- [x] Add `Bot.set_sticker_set_thumb`
- [ ] Add warning about changed parameter order for `Bot.create_new_sticker_set` and `Bot.add_sticker_to_set` to release notes

My suggestions for according new functionality:

- [ ] Add `Filters.dice` to filter messages containing a dice
- [ ] Allow `Filters.dice(1)` and `Filters.dice([1, 5])` to filters for dices with specific values
- [ ] Somehow allow to treat a dice message as text message with the corresponding emoji via `Filters.text` and `Message.effective_message`?
- [x] ~Maybe add a `description` attribute to `CommandHandler` and a method `Dispatcher.set_bot_commands` that automatically sets the bot commands by retrieving them from the registered handlers?~

**Comment by Poolitzer** [comment]
> Somehow allow to treat a dice message as text message with the corresponding emoji via Filters.text and Message.effective_message?

Thats so we can use MessageHandler and dont create DiceHandler?

**Comment by Poolitzer** [comment]
> Maybe-  add a description attribute to CommandHandler and a method Dispatcher. set_bot_commands that automatically sets the bot commands by retrieving them from the registered handlers?

Definitely needs to be opt out, and we should probably check with get commands if they already exist properly.

**Comment by Bibo-Joshi** [comment]
> > Somehow allow to treat a dice message as text message with the corresponding emoji via Filters.text and Message.effective_message?
> 
> Thats so we can use MessageHandler and dont create DiceHandler?

Well, `dice` is an attribute of `Message`, so we're fine with `MessageHandler`. But all the other "big" emojis are treated as ordinary text messages (?!) and maybe I'm not interested in the value … 

TL;DR: I think a way of receiving dice messages as text messages can be helpful

**Comment by Bibo-Joshi** [comment]
> > Maybe-  add a description attribute to CommandHandler and a method Dispatcher. set_bot_commands that automatically sets the bot commands by retrieving them from the registered handlers?
> 
> Definitely needs to be opt out, and we should probably check with get commands if they already exist properly.

I was imagining opt-in only.
Yes, checking and also for description length and number of commands before calling the API

**Comment by Poolitzer** [comment]
Upps, opt in, I meant that.

**Comment by tsnoam** [comment]
> Maybe add a description attribute to CommandHandler and a method Dispatcher.set_bot_commands that automatically sets the bot commands by retrieving them from the registered handlers?

This is an advanced feature. Shouldn't be part of releasing support to API 4.7.
At the future we can add this functioanality though unlike @Poolitzer I think that only opt-in as it's a "surprising" behaviour. It's not bound to the API and can cause undesired effects (especially since you're supposed to set all commands at once, instead of incremental updates).
Thinking about it, in the future there might be support for a more fine grained control (delete, add, update), so lets not rush into implementing something over a bleeding edge API.

**Comment by Bibo-Joshi** [comment]
>     * Thinking about it, in the future there might be support for a more fine grained control (delete, add, update), so lets not rush into implementing something over a bleeding edge API.

Sounds reasonable :+1:
